src/main/java/org/jabref/model/entry/KeywordList.java

```diff
@@ -53,15 +53,42 @@ public static KeywordList parse(String keywordString, Character delimiter, Chara
 
         KeywordList keywordList = new KeywordList();
 
-        StringTokenizer tok = new StringTokenizer(keywordString, delimiter.toString());
-        while (tok.hasMoreTokens()) {
-            String chain = tok.nextToken();
-            Keyword chainRoot = Keyword.of(chain.split(hierarchicalDelimiter.toString()));
-            keywordList.add(chainRoot);
+        List<String> tokens = splitRespectingEscapes(keywordString, delimiter);
+
+        for (String token : tokens) {
+            String[] parts = token.split(Pattern.quote(hierarchicalDelimiter.toString()));
+            Keyword chain = Keyword.of(Arrays.stream(parts).map(String::trim).toArray(String[]::new));
+            keywordList.add(chain);
         }
+
         return keywordList;
     }
 
+    private static List<String> splitRespectingEscapes(String keywordString, Character delimiter) {
+        List<String> tokens = new ArrayList<>();
+        StringBuilder currentToken = new StringBuilder();
+        boolean isEscaping = false;
+
+        for (int position = 0; position < keywordString.length(); position++) {
+            char currentChar = keywordString.charAt(position);
+
+            if (isEscaping) {
+                currentToken.append(currentChar);
+                isEscaping = false;
+            } else if (currentChar == '\\') {
+                isEscaping = true;
```

Comment:
I'm not sure whether a backslash (`\`) is a valid character in a keyword. The current implementation would remove the backslash, which might not be the intended behavior.

Reply:
Solution escaped delimiters and escaped hierarchical delimiters.